sources/packages/api/src/route-controllers/student-appeal/student-appeal.aest.controller.ts

```diff
@@ -0,0 +1,121 @@
+    private readonly workflowActionsService: WorkflowActionsService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the student appeal and its requests.
+   * @param appealId appeal id to be retrieved.
+   * @returns the student appeal and its requests.
+   */
+  @Get(":appealId/requests")
+  async getStudentAppealWithRequests(
+    @Param("appealId") appealId: number,
+  ): Promise<StudentAppealApiOutDTO> {
+    const studentAppeal =
+      await this.studentAppealService.getAppealAndRequestsById(appealId);
+    if (!studentAppeal) {
+      throw new NotFoundException("Not able to find the student appeal.");
+    }
+
+    return {
+      id: studentAppeal.id,
+      submittedDate: studentAppeal.submittedDate,
+      status: studentAppeal.status,
+      appealRequests: studentAppeal.appealRequests.map((appealRequest) => ({
+        id: appealRequest.id,
+        appealStatus: appealRequest.appealStatus,
+        submittedData: appealRequest.submittedData,
+        submittedFormName: appealRequest.submittedFormName,
+        assessedDate: appealRequest.assessedDate,
+        noteDescription: appealRequest.note?.description,
+        assessedByUserName: getUserFullName(appealRequest.assessedBy),
+      })),
+    };
+  }
+
+  /**
+   * Update all student appeals requests at once.
+   * All the appeals requests must be present with the respective
+   * approved or declined status.
+   * @param appealId appeal id to be approved/declined.
+   * @param payload appeal requests to be approved/declined.
+   * @param userToken user authentication token.
+   */
+  @Patch(":appealId/requests")
+  @ApiUnprocessableEntityResponse({
+    description:
+      `It is not possible to process the appeal approval because one of the following conditions: ` +
+      `the appeal is not in the '${StudentAppealStatus.Pending}' status; ` +
+      `the appeal is already associated with an assessment workflow; ` +
+      `some appeal request is already in a state different than '${StudentAppealStatus.Pending}'.`,
+  })
+  async approveStudentAppealRequests(
+    @Param("appealId") appealId: number,
+    @Body() payload: StudentAppealApprovalApiInDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<void> {
+    try {
+      const savedAppeal = await this.studentAppealService.approveRequests(
+        appealId,
+        payload.requests,
+        userToken.userId,
+      );
+
+      // The appeal approval will create a student assessment to be processed only
+      // if at least one request was approved, hence sometimes an appeal will not result
+      // is an assessment creation if all requests are declined.
+      if (savedAppeal.studentAssessment) {
+        await this.workflowActionsService.startApplicationAssessment(
+          savedAppeal.application.data.workflowName,
+          savedAppeal.application.id,
+          savedAppeal.studentAssessment.id,
+        );
+      }
+    } catch (error) {
+      switch (error.name) {
```

Comment:
Not a part of this PR. But IMO, we must start using unknown for error in catch blocks to eventually turn this validation **useUnknownInCatchVariables** back to true in tsconfig.json. 
May be a topic for next dev discussion.

Reply:
Yes, it would be good. Please add there 😉

Reply:
Topic added to Wiki.